wagtail/wagtailcore/tests/test_blocks.py

```diff
@@ -51,12 +64,25 @@ def test_charfield_render_with_template(self):
         self.assertEqual(html, '<h1>Hello world!</h1>')
 
     def test_charfield_render_with_template_with_extra_context(self):
-        block = blocks.CharBlock(template='tests/blocks/heading_block.html')
+        block = ContextCharBlock(template='tests/blocks/heading_block.html')
         html = block.render("Bonjour le monde!", context={
             'language': 'fr',
         })
 
-        self.assertEqual(html, '<h1 lang="fr">Bonjour le monde!</h1>')
+        self.assertEqual(html, '<h1 lang="fr">BONJOUR LE MONDE!</h1>')
+
+    def test_charfield_render_with_legacy_get_context(self):
+        block = NoExtraContextCharBlock(template='tests/blocks/heading_block.html')
+        with warnings.catch_warnings(record=True) as ws:
+            warnings.simplefilter('always')
+
+            html = block.render("Bonjour le monde!", context={
+                'language': 'fr',
+            })
+
+        self.assertEqual(len(ws), 1)
+        self.assertIs(ws[0].category, RemovedInWagtail111Warning)
+        self.assertEqual(html, '<h1>Bonjour le monde!</h1>')
```

Comment:
This should be `'<h1 lang="fr">Bonjour le monde!</h1>'`, I think...? In the case that `get_context` does not accept a `parent_context` kwarg, `render` should merge the result of `get_context(value)` into its original `context`, so that the previous behaviour is preserved.